R/lint.R

```diff
@@ -70,10 +70,18 @@ lint <- function(filename, linters = NULL, ..., cache = FALSE, parse_settings =
     return(exclude(lints, lines = lines, linter_names = names(linters), ...))
   }
 
+  file_linter_names <- names(linters)[vapply(linters, is_linter_level, logical(1L), "file")]
+  expression_linter_names <- names(linters)[vapply(linters, is_linter_level, logical(1L), "expression")]
```

Comment:
what happens to linters lacking `linter_level`, exactly?  Won't they be missing from `necessary_linters` in all cases?

Reply:
They will be in both sets. That's the `is.null() || is.na()` branch in `is_linter_level()`.

Reply:
Oh, got it.